### DISCUSSION TITLE
how to disable refetchOnWindowFocus on a page?

### DISCUSSION THREAD

**Discussion by 4lirexa** [nextauthjs/next-auth#8185] [Help] [answered]
### Question 💬

hi, i am working with live chart and data in a couple of pages of my app , the problem is anytime i switch from/to the pages that are feeded with live data , their state get updated by "refetchOnWindowFocus" and causes a flicker,
is there a way i can prevent a single page from updating the session while keeping the "refetchOnWindowFocus" enabled?

**Comment by ThangHuuVu** [comment]
not sure if I get the question correctly, but IMO the easiest way to solve is to disabled `refetchOnWindowFocus` and implement your own refetch method. Hope this help

**Reply by 4lirexa** [reply, marked as answer]
thanks for your reply, the problem was the "refetchOnWindowFocus" caused a re-render on browser tab focus , and the re-render caused state update in component and reseted the chart and XTERM data, so a connected websocket connection would get disconneted and reconnect again, any ways i ended up with exporting a prop on the components i want to disable "refetchOnWindowFocus" on and a logic in _app.jsx page,
here is the code:
on component:
`Home.shouldEnableRefetch = false;`
on _app.jsx:
`const shouldEnableRefetch = Component?.shouldEnableRefetch == false ? false : true

<SessionProvider
              session={session}
              refetchInterval={5 * 60}
              refetchOnWindowFocus={refetchOnWinEnabled}
            >
{Component}
</SessionProvider>`
